code/game/machinery/camera/camera.dm

```diff
@@ -388,3 +388,18 @@
 	if(cameranet && get_turf(src) != prev_turf)
 		cameranet.updatePortableCamera(src)
 		prev_turf = get_turf(src)
+
+/obj/machinery/camera/get_vision_impairments(mob/user)
+	. = list()
+	if(view_range == short_range) //unfocused
+		. += global_hud.darkMask
+
+/obj/machinery/camera/update_remote_sight(mob/living/user)
+	user.see_invisible = SEE_INVISIBLE_LIVING //can't see ghosts through cameras
+	if(isXRay())
+		user.sight |= (SEE_TURFS|SEE_MOBS|SEE_OBJS)
+		user.see_in_dark = max(user.see_in_dark, 8)
+	else
+		user.sight = 0
+		user.see_in_dark = 2
+	return 1
```

Comment:
Wearing thermal goggles and looking through a camera no longer lets you see mobs through walls. If the camera has been upgraded with xray, you actually see like you had xray mutation while using the camera.